site/src/components/ui/multiple-selector.tsx

```diff
@@ -0,0 +1,649 @@
+			if (!emptyIndicator) return undefined;
+
+			// For async search that showing emptyIndicator
+			if (onSearch && !creatable && Object.keys(options).length === 0) {
+				return (
+					<CommandItem value="-" disabled>
+						{emptyIndicator}
+					</CommandItem>
+				);
+			}
+
+			return <CommandEmpty>{emptyIndicator}</CommandEmpty>;
+		}, [creatable, emptyIndicator, onSearch, options]);
+
+		const selectables = React.useMemo<GroupOption>(
+			() => removePickedOption(options, selected),
+			[options, selected],
+		);
+
+		/** Avoid Creatable Selector freezing or lagging when paste a long string. */
+		const commandFilter = React.useCallback(() => {
+			if (commandProps?.filter) {
+				return commandProps.filter;
+			}
+
+			if (creatable) {
+				return (value: string, search: string) => {
+					return value.toLowerCase().includes(search.toLowerCase()) ? 1 : -1;
+				};
+			}
+			// Using default filter in `cmdk`. We don't have to provide it.
+			return undefined;
+		}, [creatable, commandProps?.filter]);
+
+		return (
+			<Command
+				ref={dropdownRef}
+				{...commandProps}
+				onKeyDown={(e) => {
+					handleKeyDown(e);
+					commandProps?.onKeyDown?.(e);
+				}}
+				className={cn(
+					"h-auto overflow-visible bg-transparent",
+					commandProps?.className,
+				)}
+				shouldFilter={
+					commandProps?.shouldFilter !== undefined
+						? commandProps.shouldFilter
+						: !onSearch
+				} // When onSearch is provided, we don't want to filter the options. You can still override it.
+				filter={commandFilter()}
+			>
+				{/* biome-ignore lint/a11y/useKeyWithClickEvents: <explanation> */}
+				<div
+					className={cn(
+						"min-h-9 rounded-md border border-solid border-border text-sm pr-3",
+						{
+							"pl-3 py-1": selected.length !== 0,
+							"cursor-text": !disabled && selected.length !== 0,
+						},
+						className,
+					)}
+					onClick={() => {
+						if (disabled) return;
+						inputRef?.current?.focus();
+					}}
+				>
+					<div className="relative flex flex-wrap gap-1 items-center">
+						{selected.map((option) => {
+							return (
+								<Badge
+									key={option.value}
+									className={cn(
+										"data-[disabled]:bg-muted-foreground data-[disabled]:text-muted data-[disabled]:hover:bg-muted-foreground",
+										"data-[fixed]:bg-muted-foreground data-[fixed]:text-muted data-[fixed]:hover:bg-surface-secondary",
+										badgeClassName,
+									)}
+									data-fixed={option.fixed}
+									data-disabled={disabled || undefined}
+								>
+									{option.label}
+									<button
+										type="button"
+										className={cn(
+											"ml-1 pr-0 rounded-full bg-transparent border-none outline-none focus:ring-2 focus:ring-surface-invert-primary focus:ml-2.5 focus:pl-0 cursor-pointer",
+											(disabled || option.fixed) && "hidden",
+										)}
+										onKeyDown={(e) => {
+											if (e.key === "Enter") {
+												handleUnselect(option);
+											}
+										}}
```

Comment:
This shouldn't be necessary. The HTML spec makes sure that an `onClick` handler is also automatically wired up to the `onKeyDown` event for Enter and the `onKeyUp` event for Space

The whole thing can be removed

Reply:
When I test this by removing the `onKeyDown` handler by tabbing to an item in the multi-select and pressing enter, I am no longer able to unselect/remove the item from the multi-select component.

Reply:
Okay, in that case, they might be doing some things that break from the spec. I'm okay with keeping it – I would just like a comment explaining that it is necessary

Reply:
@Parkreiner Can you refer to where in the HTML spec you found this?

Reply:
btw if I remove onKeyDown, the key that actually works in this case is space instead of enter